homeassistant/components/time_date/sensor.py

```diff
@@ -84,6 +79,18 @@ def icon(self):
             return "mdi:calendar"
         return "mdi:clock"
 
+    async def async_added_to_hass(self) -> None:
+        """Set up next update."""
+        self.unsub = async_track_point_in_utc_time(
+            self.hass, self.point_in_time_listener, self.get_next_interval()
+        )
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Cancel next update."""
+        if self.unsub:
+            self.unsub()
+            self.unsub = None
```

Comment:
```suggestion
        self.async_on_remove(
            async_track_point_in_utc_time(
                self.hass, self.point_in_time_listener, self.get_next_interval()
            )
        )
```

Reply:
Ah, nice! I wasn't aware of `async_on_remove`.

Reply:
But wait. I don't think that will work. First, because `async_track_point_in_utc_time()` will effectively remove it itself when it runs the action. And second because `async_track_point_in_utc_time()` is also called in `self.point_in_time_listener()` every time the sensor updates. Using `async_on_remove` would constantly add to the list "un-subcribers" that almost immediately become stale.

Reply:
oh wait, yes you're right. Sorry. I was thinking of track interval.